iModelCore/BeSQLite/BeSQLite.cpp

```diff
@@ -3167,23 +3188,22 @@ void Db::FlushPageCache()
 +---------------+---------------+---------------+---------------+---------------+------*/
 bool Db::ColumnExists(Utf8CP tableName, Utf8CP columnName) const
     {
-    Statement sql;
-    return BE_SQLITE_OK == sql.TryPrepare(*this, SqlPrintfString("SELECT [%s] FROM %s", columnName, tableName));
+    bvector<Utf8String> columns;
+
+    GetColumns(columns, tableName);
+    return columns.end() != std::find_if(columns.begin(), columns.end(), [columnName](Utf8StringCR col) { return col.EqualsIAscii(columnName); });
     }
 
 /*---------------------------------------------------------------------------------**//**
 * @bsimethod
 +---------------+---------------+---------------+---------------+---------------+------*/
 bool Db::GetColumns(bvector<Utf8String>& columns, Utf8CP tableName) const
     {

```

Comment:
You might also want to parse tableName and look for dot, as table name could be qualified by tablespace.

In this case, `PRAGMA table_info(?)` if table space as not given. Otherwise, use  `PRAGMA %s.table_info(?)`

and `columns.push_back(statement->GetValueText(0));` need to change to `columns.push_back(statement->GetValueText(`));`

Reply:
addressed